### PR TITLE
RetroAchievements - Fixed Leaderboard Scored message format

### PR DESCRIPTION
**Pull request by LillyJadeKatrin**
Refactored the message displayed when a leaderboard is submitted to so that it properly generates and uses a FormattedValue.

### CODE REVIEW

**Comment by AdmiralCurtiss** [comment]
Is this guaranteed to return a nullterminated string? I'm not seeing anything indicating that in the [documentation](https://github.com/RetroAchievements/rcheevos/wiki/rc_runtime_format_lboard_value). In fact, by that I would expect:

```
FormattedValue value;
int length = rc_runtime_format_lboard_value(value.data(), FORMAT_SIZE, runtime_event->value, m_game_data.leaderboards[ix].format);
std::string_view str(value.data(), static_cast<size_t>(length));
OSD::AddMessage(fmt::format("Scored {} on leaderboard: {}", str, m_game_data.leaderboards[ix].title),
```

**Comment by LillyJadeKatrin** [comment]
Under the hood, every call to both rc_runtime_format_lboard_value and rc_runtime_format_achievement_measured calls sprintf on a string literal, which to my understanding should always nullterminate.

**Comment by LillyJadeKatrin** [comment]
Also, the maximum returned size of a formatted string can be 21 characters, and I was recommended to use 24 as a nice round number, so there's space allocated for the terminator.

File: Source/Core/Core/AchievementManager.h [hunk 1]
```diff
@@ -20,6 +20,8 @@
 #include "Common/WorkQueueThread.h"
 
 using AchievementId = u32;
+constexpr size_t FORMAT_SIZE = 24;
```
Comment (lioncash) [inline Source/Core/Core/AchievementManager.h:23]: Might want to make this a more unique name. This is pretty generic for something that's in the global namespace

File: Source/Core/Core/AchievementManager.cpp [hunk 2]
```diff
@@ -716,7 +716,10 @@ void AchievementManager::HandleLeaderboardTriggeredEvent(const rc_runtime_event_
   {
     if (m_game_data.leaderboards[ix].id == runtime_event->id)
     {
-      OSD::AddMessage(fmt::format("Scored {} on leaderboard: {}", runtime_event->value,
+      FormattedValue value;
```
Comment (lioncash) [inline Source/Core/Core/AchievementManager.cpp:719]:
```suggestion
      FormattedValue value{};
```

File: Source/Core/Core/AchievementManager.cpp [hunk 3]
```diff
@@ -716,7 +716,10 @@ void AchievementManager::HandleLeaderboardTriggeredEvent(const rc_runtime_event_
...
+      rc_runtime_format_lboard_value(value.data(), FORMAT_SIZE, runtime_event->value,
```
Comment (lioncash) [inline Source/Core/Core/AchievementManager.cpp:720]:
```suggestion
      rc_runtime_format_lboard_value(value.data(), value.size(), runtime_event->value,
```

File: Source/Core/Core/AchievementManager.cpp [hunk 4]
```diff
@@ -716,7 +716,10 @@ void AchievementManager::HandleLeaderboardTriggeredEvent(const rc_runtime_event_
   {
     if (m_game_data.leaderboards[ix].id == runtime_event->id)
     {
-      OSD::AddMessage(fmt::format("Scored {} on leaderboard: {}", runtime_event->value,
+      FormattedValue value{};
+      rc_runtime_format_lboard_value(value.data(), static_cast<int>(value.size()),
+                                     runtime_event->value, m_game_data.leaderboards[ix].format);
+      OSD::AddMessage(fmt::format("Scored {} on leaderboard: {}", value.data(),
```
Comment (iwubcode) [inline Source/Core/Core/AchievementManager.cpp:722]: If we are concerned with null-termination, we could at least clamp if we don't find it in the buffer.  Something like:

```
if (std::find(value.begin(), value.end(), '\0') == value.end())
{
  OSD::AddMessage(fmt::format("Scored {} on leaderboard: {}", std::string_view{value.data(), value.size()}, ...
}
else
{
  OSD::AddMessage(fmt::format("Scored {} on leaderboard: {}", value.data(), ...
}
```

**Review by iwubcode — APPROVED** [review]

**Pull request merged**